JitsiMeetJS.ts

```diff
@@ -495,6 +509,104 @@ export default {
+     */
+    async joinConference(
+            roomName: string,
+            appId: string = '',
+            token: string | null = null,
+            options: IJoinConferenceOptions = {}): Promise<JitsiConference> {
+        const d = new Deferred();
+        let connectionOptions = options.connectionOptions ?? {};
+
+        // Provide a solid default config in case of JaaS.
+        if (appId.startsWith('vpaas-magic-cookie')) {
+            // Auto-initialize the library.
+            const autoInit = Symbol.for('kLJMAutoInit');
+
+            if (typeof window[autoInit] === 'undefined') {
+                window[autoInit] = true;
+
+                // TODO(saghul): review rtcstats initialization.
+                this.init({
+                    analytics: {
+                        rtcstatsEnabled: true,
+                        rtcstatsStoreLogs: true,
+                        rtcstatsSendSdp: true,
+                    },
+                    enableAnalyticsLogging: true,
+                });
+            }
+
+            const useStage = options.jaas?.useStaging ?? false;
+            const jaasDomain = useStage ? 'staging.8x8.vc' : '8x8.vc';
+            const opts = {
+                hosts: {
+                    domain: jaasDomain,
+                    muc: `conference.${appId}.${jaasDomain}`
+                },
+                conferenceRequestUrl: `https://${jaasDomain}/${appId}/conference-request/v1?room=${roomName}`,
+                serviceUrl: `wss://${jaasDomain}/${appId}/xmpp-websocket?room=${roomName}`,
+                websocketKeepAliveUrl: `https://${jaasDomain}/${appId}/_unlock?room=${roomName}`,
+                analytics: {
+                    rtcstatsEnabled: true,
+                    rtcstatsEndpoint: `wss://rtcstats-server-${useStage ? 'pilot' : '8x8'}.jitsi.net/`,
+                },
+            };
+
+            connectionOptions = merge(connectionOptions, opts);
+        }
+
+        const conn = new JitsiConnection(appId, token, connectionOptions);
+
+        function cleanupListeners() {
+            conn.removeEventListener(
+                JitsiConnectionEvents.CONNECTION_ESTABLISHED, onConnectionEstablished);
+            conn.removeEventListener(
+                JitsiConnectionEvents.CONNECTION_FAILED, onConnectionFailed);
+        }
+
+        function onConnectionEstablished() {
+            cleanupListeners();
+
+            const conf = conn.initJitsiConference(roomName, options.conferenceOptions ?? {});
+
+            d.resolve(conf);
+
+            // Make sure this runs after the promise was resolved so that local track events can be
+            // listened to.
+            queueMicrotask(() => {
+                for (const track of options.tracks || []) {
+                    conf.addTrack(track);
+                }
+
+                conf.join();
+
+                // TODO(saghul): set some sender / receiver constraints?
```

Comment:
@jallamsetty1 WDYT? IMHO our current default of 180p is bad, it makes for a bad first impression. How about defaulting to 720?

Since that might be disruptive we could set the constraints here, when using this "simplified" API.

Reply:
This is a bit tricky. We do not have the source info at this point so the client will have to set the default constraints to 720p. That means when a user joins the call, the bridge will ask all the other endpoints in the call to start sending 720p even if there are being viewed or not. This could create a lot of unnecessary traffic and overload the bridges.
Even if the application updates the receiver constraints, there may be some delay and depending on how they implement it, if they don't reset the default constraints back to 0, the bridge may end up sending a lot of unnecessary traffic down to the endpoints.
That is why the default is set to 180p in the bridge. Maybe better documentation would help here?

Reply:
The status quo is bad and documentation won't change that. Imagine this, you are evaulating video products and see some upscaled 180p stream looking terrible. Would you dig deeper?

Perhaps we can find some middle gound with 360p, or 720p but limiting lastN?

Reply:
Some data from other comparable libraries:

- mediaSoup: sends the highest layer if not configured to a specific one
- janus: same as mediasoup
- livekit: it has an adaptive mode which will select the receive resolution based on the attached element size and visibility, this is nice!
- twilio: same as livekit, when "adaptive simulcast" mode is used

IMHO any option is better than our current one.

Reply:
I agree that we need a better one. I voted for something similar to what livekit and Twilio do when we first implemented receiver constraints but product wanted the application to control what is being requested based on the layout instead of the library doing it automatically. We need to involve product and have a much broader discussion.